hazelcast/src/main/java/com/hazelcast/sql/impl/type/QueryDataTypeUtils.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.type;
+
+import com.hazelcast.sql.impl.type.converter.BigDecimalConverter;
+import com.hazelcast.sql.impl.type.converter.BigIntegerConverter;
+import com.hazelcast.sql.impl.type.converter.CalendarConverter;
+import com.hazelcast.sql.impl.type.converter.CharacterConverter;
+import com.hazelcast.sql.impl.type.converter.Converter;
+import com.hazelcast.sql.impl.type.converter.Converters;
+import com.hazelcast.sql.impl.type.converter.DateConverter;
+import com.hazelcast.sql.impl.type.converter.InstantConverter;
+import com.hazelcast.sql.impl.type.converter.OffsetDateTimeConverter;
+import com.hazelcast.sql.impl.type.converter.StringConverter;
+import com.hazelcast.sql.impl.type.converter.ZonedDateTimeConverter;
+
+/**
+ * Utility methods for SQL data types.
+ */
+public final class QueryDataTypeUtils {
+    /** 12 (hdr) + 12 (fields) + 12 (arr) + 4 (arr len) + 16 (eight chars) */
+    public static final int TYPE_LEN_VARCHAR = 12 + 4 + 8 + 12 + 4 + 16;
```

Comment:
This might be too verbose, especially for complex types, such as temporal classes. I would leave it as is without explicitly typing the fields names of classes from the Java standard library.